azuredevops/internal/service/taskagent/resource_variable_group.go

```diff
@@ -108,20 +108,20 @@ func ResourceVariableGroup() *schema.Resource {
 							Type:          schema.TypeString,
 							Optional:      true,
 							Default:       "",
-							ConflictsWith: []string{vgKeyVault},
+							ConflictsWith: []string{vgKeyVault, vgIsSecret, secretVgValue},
```

Comment:
`value, secret_value, is_secret` all have default values, the constraint will always  be valid, internal check will always failed

Reply:
You are absolutely right. Moreover, referencing objects in the Set is [not supported](https://github.com/hashicorp/terraform-plugin-sdk/issues/71) in `ConflictsWith` at the moment. I have another idea how to achieve this but it will be more dangerous and deserves a new PR.